docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,216 @@
+---
+title: "Train a Model with a Custom Python Training Script"
+linkTitle: "Use a Custom Training Script"
+weight: 90
+type: "docs"
+tags: ["data management", "ml", "model training"]
+description: "How to write and upload a custom machine learning model training script to the Viam registry and use it to submit training jobs."
+no_service: true
+# SME: Tahiya S.
+---
+
+You can train your classification and object detection models on the [Viam app](https://app.viam.com), where the trained model is output as a TFLite model.
+See [Train a Model](/services/ml/train-model/) for more information.
+However, you can now also upload a custom training script to the [Viam registry](https://app.viam.com/registry/) to train your ML model with an output type of PyTorch, Tensorflow, TFLite, ONNX, or any other model artifacts.
+A flexible training interface like this allows you to upload any machine learning training code for your use-case.
+
+Follow this guide to create, upload, and submit your Python script that loads a training dataset, trains an ML model, and produces a model artifact.
+
+## Create a training script
+
+You must save your training script in the `tar.gz` format to run in the Viam ML training service.
+Follow this guide to prepare a Python source distribution `tar.gz` file with the code for training your model.
+
+### Create entrypoint file <file>model/training.py</file>
+
+To start, create an entrypoint file that contains the main logic in a file called <file>training.py</file> in a folder called <file>model</file>.
+
+Add the following logic to your code in <file>training.py</file>:
+
+{{%expand "Step 1: Parse command line inputs" %}}
+
+This script must take as command line inputs a `dataset_file` and `model_output_directory`.
+The `dataset_file` is a file containing the data and metadata for the training job and the `model_output_directory` is the location where the user-defined model artifacts (the TFLite, PyTorch, TF, ONNX files) produced from training are saved to.
+Parse these arguments in your training script as follows:
+
+```python {class="line-numbers linkable-line-numbers" data-line="364"}
+# This parses the required args for running the training script.
+# You can use the data JSON to parse the images and
```

Comment:
what does this mean? How would I used the data JSON to parse images and annotations? I think that needs to be a separate step

Reply:
yeah it is a separate step already the parsing classification and bbox helper functions that come next! Do you think this still needs to be clearer or if its ok as like an introduction as to why we're parsing these values? @npentrel